src/lib/ares__close_sockets.c

```diff
@@ -97,6 +97,16 @@ void ares__check_cleanup_conn(const ares_channel_t     *channel,
     do_cleanup = ARES_TRUE;
   }
 
+  /* If the associated server has a serious failure, close it out. Resetting
+   * the connection (and specifically the source port number) can help resolve
+   * situations where packets are being dropped.
+   */
+  if (channel->optmask & ARES_OPT_SERVER_FAILOVER &&
+      channel->server_serious_fail_limit > 0 &&
+      conn->server->consec_failures >= channel->server_serious_fail_limit) {
+    do_cleanup = ARES_TRUE;
+  }
+
```

Comment:
This really only has an impact if someone is using ARES_FLAG_STAYOPEN. Honestly, I think this should just be:
```
if (conn->server->consec_failures) {
```
and call it a day.  I don't see a need to hit some limit to kill the connection in this edge case, if there's a failure, kill it.

Reply:
Ok, makes sense and seems sensible. I'll make this change.